src/hooks/use-breakpoint.ts

```diff
@@ -0,0 +1,85 @@
+'use client'
+
+import { useEffect, useMemo, useState } from 'react'
+
+import { type Breakpoint, breakpoints } from '~/styles'
+
+type UseIsBreakpointProps = {
+  breakpoint: Breakpoint
+}
+
+/**
+ * Custom hook to determine if the current window width matches a specified breakpoint.
+ *
+ * @param {UseIsBreakpointProps} props - The properties object.
+ * @param {Breakpoint} props.breakpoint - The breakpoint to match against.
+ * @returns {boolean} - A boolean indicating if the current window width matches the specified breakpoint.
+ */
+export const useIsBreakpoint = ({ breakpoint }: UseIsBreakpointProps) => {
```

Comment:
In a server-side rendered (SSR) environment, 'window' is not available during initial render.
Attempting to access 'window' directly can cause runtime errors in SSR contexts.
By adding:
`if (typeof window === 'undefined') {
    return false // or some initial value like `false`
}`
we prevent this code from running on the server, ensuring it only runs in the browser.